core/src/main/java/org/apache/iceberg/BaseScan.java

```diff
@@ -165,6 +170,19 @@ public ThisT includeColumnStats() {
     return newRefinedScan(table, schema, context.shouldReturnColumnStats(true));
   }
 
+  @Override
+  public ThisT includeColumnStats(Collection<String> requestedColumns) {
+    return newRefinedScan(
+        table,
+        schema,
+        context
+            .shouldReturnColumnStats(true)
```

Comment:
Supporting both `shouldReturnColumnStats()` and `columnsToKeepStats()` will only be needed if we decide to go with the utility for copying. If `null` is a valid value in `ContentFile$copyWithStats`, I think having just a set of field IDs to keep stats for would be sufficient.

Reply:
Let's talk about this.
My main arguments will be:
- Backward compatibility - no need to change the API
- No hidden meaning for specific list values (null, empty)